server/sonar-server/src/test/java/org/sonar/server/issue/ws/SearchActionMediumTest.java

```diff
@@ -466,15 +466,36 @@ public void filter_by_assigned_to_me() throws Exception {
     session.commit();
     tester.get(IssueIndexer.class).indexAll();
 
-    MockUserSession.set().setLogin("john");
+    MockUserSession.set().setLogin("john-bob.polop");
     wsTester.newGetRequest(IssuesWs.API_ENDPOINT, SearchAction.SEARCH_ACTION)
       .setParam("resolved", "false")
       .setParam("assignees", "__me__")
       .setParam(WebService.Param.FACETS, "assignees,assigned_to_me")
       .execute()
       .assertJson(this.getClass(), "filter_by_assigned_to_me.json", false);
+  }
+
+  @Test
+  public void filter_by_assigned_to_me_unauthenticated() throws Exception {
+    ComponentDto project = insertComponent(ComponentTesting.newProjectDto("ABCD").setKey("MyProject"));
+    setDefaultProjectPermission(project);
+    ComponentDto file = insertComponent(ComponentTesting.newFileDto(project, "BCDE").setKey("MyComponent"));
+    RuleDto rule = newRule();
+    IssueDto issue1 = IssueTesting.newDto(rule, file, project)
+      .setStatus("OPEN")
+      .setKee("82fd47d4-b650-4037-80bc-7b112bd4eac2")
+      .setAssignee("john");
+    IssueDto issue2 = IssueTesting.newDto(rule, file, project)
+      .setStatus("OPEN")
+      .setKee("7b112bd4-b650-4037-80bc-82fd47d4eac2")
+      .setAssignee("alice");
+    IssueDto issue3 = IssueTesting.newDto(rule, file, project)
+      .setStatus("OPEN")
+      .setKee("82fd47d4-4037-b650-80bc-7b112bd4eac2");
+    db.issueDao().insert(session, issue1, issue2, issue3);
+    session.commit();
+    tester.get(IssueIndexer.class).indexAll();
```

Comment:
So you need it !
As UserSession is static, it should be defined in every test case. You're lucky that the test is green, but following in which order test cases are executed, this test case might fail.